### PR TITLE
Add advance to date and time

### PR DESCRIPTION
**Pull request by timgaleckas**
as extended by activesupport

### CODE REVIEW

File: lib/activesupport/all/activesupport.rbi [hunk 1]
```diff
@@ -548,6 +556,10 @@ class Time
   sig { returns(Time) }
   def beginning_of_day; end
 
+  sig { params(options: T::Hash[Symbol, Integer]).returns(Time) }
+  def advance(options)
```
Comment (connorshea) [inline lib/activesupport/all/activesupport.rbi]: These should be formatted as one line, like `def advance(options); end`

File: lib/activesupport/all/activesupport.rbi [hunk 2]
```diff
@@ -539,6 +539,14 @@ class ActiveSupport::TimeWithZone
   def at_middle_of_day; end
 end
 
+# defines some of the methods at https://github.com/rails/rails/blob/v6.0.0/activesupport/lib/active_support/core_ext/date
+# this is not a complete definition!
+class Date
+  sig { params(options: T::Hash[Symbol, Integer]).returns(Date) }
+  def advance(options)
```
Comment (connorshea) [inline lib/activesupport/all/activesupport.rbi]: This too

File: lib/activesupport/all/activesupport.rbi (hunk 1 above)
Comment (timgaleckas) [reply lib/activesupport/all/activesupport.rbi]: Yup. Fixed. Thanks.

File: lib/activesupport/all/activesupport.rbi (hunk 2 above)
Comment (timgaleckas) [reply lib/activesupport/all/activesupport.rbi]: Yup. Fixed. Thanks.

**Pull request merged**